compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -225,6 +225,11 @@ public boolean checkForUniqueSymbol(Location pos, SymbolEnv env, BSymbol symbol)
             return true;
         }
 
+        if (symbol.tag == SymTag.XMLNS &&
+                !(((BXMLNSSymbol) foundSym).compUnit.value.equals(((BXMLNSSymbol) symbol).compUnit.value))) {
```

Comment:
Here, why we need to check whether the compile units are equal?

Reply:
XMLNS prefix is valid for a single file. Thus we need to ensure that the symbol we retrieved belongs to the same file.